File: 209-minimum-size-subarray-sum/minimum-size-subarray-sum.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int minlen = INT_MAX;
        int i=0, j=0;
        int sum = 0;
        while (j < nums.size()){
            sum += nums[j];
            if (sum >= target){
                while(i <= j && sum >= target){
                    minlen = min(minlen, j-i+1);
                    sum-=nums[i];
                    i++;
                }
            }
            j++;
        }
        return (minlen == INT_MAX) ? 0 : minlen;
    }
};

File: 209-minimum-size-subarray-sum/minimum-size-subarray-sum.go
func minSubArrayLen(target int, nums []int) int {
    var min_len, sum, j int = len(nums)+1, 0, 0
    for i := 0; i < len(nums); i++ {
        sum += nums[i]

        win_len := i - j + 1

        for j <= i && sum >= target {
            min_len = min(min_len, win_len)
            sum -= nums[j]
            j++
            win_len = i - j + 1
        }
    }

    if min_len == len(nums) + 1 { return 0 }

    return min_len
}